src/qibo/backends/pytorch.py

```diff
@@ -195,6 +195,11 @@ def calculate_matrix_exp(self, a, matrix, eigenvectors=None, eigenvalues=None):
         ud = self.np.conj(eigenvectors).T
         return self.np.matmul(eigenvectors, self.np.matmul(expd, ud))
 
+    def calculate_matrix_power(self, matrix, power):
+        copied = self.to_numpy(self.np.copy(matrix))
+        copied = super().calculate_matrix_power(copied, power)
+        return self.cast(copied, dtype=copied.dtype)
```

Comment:
This probably brakes gradient flow with `pytorch`, but unfortunately there is no way of avoiding it unless we implement fractional exponentiation on our side...

Reply:
The copy is needed specifically because `torch` doesn't allow the `scipy` function to work on a `tensor` that requires gradients. That's why we need to calculate the power on a copy. I'm not sure if the process of copying breaks the graph of the original tensor. Moreover, I'm not expecting this function to be used on something that requires optimization, unless there's some niche case that I'm not aware of.

Reply:
It's not much the copy, but rather the conversion to numpy. scipy functions fail with torch tensors that require gradients because they internally try to convert them to numpy arrays, which is not possible (unless you force it) when they have `requires_grad=True`. But yeah, regardless, in this case there's not much you can do except implementing the whole fractional power exponentiation with torch natives. However, I would maybe add a comment or docstring remind us that this breaks gradient flow.